documentation/how-to/archival.md

```diff
@@ -0,0 +1,130 @@
+```
+
+## Configure S3 Integrator and Relate to Temporal K8s
+
+Archival can be configured on any S3-compatible storage. S3 access and
+configurations are managed with the
+[Charmed S3 Integrator](https://charmhub.io/s3-integrator/configure):
+
+```bash
+juju config s3-integrator \
+    endpoint="https://s3.us-west-2.amazonaws.com" \
+    bucket="test-bucket-1" \
+    path="/temporal-archival" \
+    region="us-west-2"
+
+juju run s3-integrator/leader sync-s3-credentials access-key=<access-key> secret-key=<secret-key> --wait 1m
+juju relate temporal-k8s s3-integrator
+```
+
+Wait until the application is ready - when it is ready,
+`juju status --relations` will show:
+
+```
+Model           Controller           Cloud/Region        Version  SLA          Timestamp
+temporal-model  temporal-controller  microk8s/localhost  3.1.5    unsupported  12:35:24+03:00
+
+App                 Version   Status    Scale  Charm                Channel    Rev  Address         Exposed  Message
+s3-integrator                 active       1   s3-integrator        stable      13  10.152.183.171  no
+postgresql-k8s      14.7      active       1   postgresql-k8s       14/stable   73  10.152.183.250  no       Primary
+temporal-admin-k8s            active       1   temporal-admin-k8s   stable       4  10.152.183.21   no
+temporal-k8s                  active       1   temporal-k8s         stable       9  10.152.183.191  no
+
+Unit                   Workload   Agent  Address      Ports  Message
+s3-integrator/0*       active     idle   10.1.232.2
+postgresql-k8s/0*      active     idle   10.1.232.66         Primary
+temporal-admin-k8s/0*  active     idle   10.1.232.71
+temporal-k8s/0*        active     idle   10.1.232.64
+
+Integration provider               Requirer                           Interface            Type     Message
+postgresql-k8s:database            openfga-k8s:database               postgresql_client    regular
+postgresql-k8s:database            temporal-k8s:db                    postgresql_client    regular
+postgresql-k8s:database            temporal-k8s:visibility            postgresql_client    regular
+postgresql-k8s:database-peers      postgresql-k8s:database-peers      postgresql_peers     peer
+postgresql-k8s:restart             postgresql-k8s:restart             rolling_op           peer
+postgresql-k8s:upgrade             postgresql-k8s:upgrade             upgrade              peer
+s3-integrator:s3-credentials       temporal-k8s:s3-parameters         s3                   regular
+s3-integrator:s3-integrator-peers  s3-integrator:s3-integrator-peers  s3-integrator-peers  peer
+temporal-admin-k8s:admin           temporal-k8s:admin                 temporal             regular
+temporal-admin-k8s:peer            temporal-admin-k8s:peer            temporal-admin       peer
+temporal-k8s:peer                  temporal-k8s:peer                  temporal             peer
+```
+
+Note: Make sure to observe the the output of `juju debug-log` to ensure that
```

Comment:
I wonder if there is no better place to surface this info. I know the Postgresql charm shows info on `juju status` when the s3 relation fails but that might add a lot of complexity to your charm considering the charm can still operate with a failed relation to the s3-integrator and the status message can be overwritten.

Reply:
I agree, for the PostgreSQL charm you would know if everything is working fine if you attempt to run a backup action. In this case, you would know that everything is working if you run a sample workflow and see that events are being backed up to the S3 storage. I've tried to cover the common cases (e.g. missing or failed to create bucket) in the `_validate` method.